src/control/lib/daos/pool.go

```diff
@@ -87,8 +96,117 @@ type (
 		Free      uint64           `json:"free"`
 		MediaType StorageMediaType `json:"media_type"`
 	}
+
+	// PoolQueryMask implements a bitmask for pool query options.
+	PoolQueryMask C.uint64_t
 )
 
+// DefaultPoolQueryMask defines the default pool query mask.
+const DefaultPoolQueryMask = PoolQueryMask(^uint64(0) &^ (C.DPI_ENGINES_ENABLED | C.DPI_ENGINES_DISABLED))
```

Comment:
It occurs to me, after I maybe got a little carried away with allowing the mask to be set via helper methods, that it might be best to define a `HealthOnlyQueryMask` here so that various users of the API don't have to implement their own versions of it. Alternatively, add a `SetHealthOnly()` helper method? Might be a little over-engineered...

Reply:
we've also done this elsewhere in dmg so that might be useful as a reference as well just to try to keep some consistency in how we manage query state filtering in command handling

Reply:
Yeah, I'm not sure what I was thinking with the previous approach... Maybe I was a little fixated on the narrow use case of handling CLI flags. Anyhow, I'm much happier with what I just pushed. I went with `SetAll()`, `ClearAll()`, `SetOptions()`, `ClearOptions()`. It's a bitmask, so the set/clear verbs seemed to make sense to me. 

Not sure what the distinction between add and set might be. I guess set might be exclusive? IMO, if someone wants to exclusively set some options, it would be clearer to clear everything first and then set whatever.

Reply:
Yeah, I was thinking set as override/reset. `ClearAll()` gets the job done. Thanks.